drivers/serial/uart_sam0.c

```diff
@@ -925,21 +925,22 @@ static const struct uart_driver_api uart_sam0_driver_api = {
 #endif
 };
 
-#if CONFIG_UART_INTERRUPT_DRIVEN || CONFIG_UART_ASYNC_API
-#define DT_ATMEL_SAM0_UART_SERCOM_IRQ(n, m) DT_ATMEL_SAM0_UART_SERCOM_ ## n ## _IRQ_ ## m
-#define DT_ATMEL_SAM0_UART_SERCOM_IRQ_PRIORITY(n, m) DT_ATMEL_SAM0_UART_SERCOM_ ## n ## _IRQ_ ## m ## _PRIORITY
+#define UART(n)		DT_NODELABEL(sercom##n)
+#define HAS_UART(n)	DT_NODE_HAS_COMPAT(UART(n), atmel_sam0_uart)
 
+#if CONFIG_UART_INTERRUPT_DRIVEN || CONFIG_UART_ASYNC_API
```

Comment:
This should be:
`#if defined(CONFIG_UART_INTERRUPT_DRIVEN) || defined(CONFIG_UART_ASYNC_API)`

Reply:
Done!